mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -192,6 +195,38 @@ def stem_word(self, word):
         """
         return self.stemmer.stem_word(word)
 
+    def tojson(self):
+        """
+        Method defined to obtain recursive JSON representation of a TextPreparationPipeline.
+
+        Args:
+            None.
+
+        Returns:
+            JSON representation of TextPreparationPipeline (dict) .
+        """
+        return {
+            "version": get_mm_version(),
+            "language": self.language,
+            "preprocessors": self.preprocessors,
+            "normalizers": self.normalizers,
+            "tokenizer": self.tokenizer,
+            "stemmer": self.stemmer
+        }
+
+    def get_hashid(self):
+        """
+        Method defined to obtain Hash value of TextPreparationPipeline.
+
+        Args:
+            None.
+
+        Returns:
+            256 character hash representation of current TextPreparationPipeline config (str) .
+        """
+        string = json.dumps(self, cls=TPPJSONEncoder, sort_keys=True)
```

Comment:
Nice catch on sorting the keys.  FYI, Python dicts keep items in insertion order since 3.6, but it's not officially documented until 3.7, so sorting these is good form since we support 3.6.